app/app.js

```diff
@@ -125,7 +125,27 @@ app.get('/api/user/:userName/reports', (req, res) => {
         }, function (err) {
             res.send(err);
         })
-  });  
+  });
+  
+  app.get('/api/executeImmediate/', (req, res) => {
+    var visualizationid= req.query.visualizationid;
+    result = jobs.executeImmediate(visualizationid);
+    result.then(function (result) {
+            res.send(result);
+        }, function (err) {
+            res.send(err);
+        })
+  });
+
+  app.get('/api/jobLogs/', (req, res) => {
+    var visualizationid= req.query.visualizationid;
+    result = jobs.jobLogs(visualizationid);
```

Comment:
there is no need of this line u can write the code as below

jobs.jobLogs(visualizationid).then